### PR TITLE
soc/riscv/sifive-freedom/fe310: use correct SYS_CLOCK_HW_CYCLES_PER_SEC

### PR DESCRIPTION
**Pull request by fkokosinski**
This PR introduces several changes that allow the SYS_CLOCK_HW_CYCLES_PER_SEC value to revert back to its correct value, as reported by #44432 and #44856, while making tests still pass on the HiFive1 board. The previously-failing tests were:
* tests/kernel/sleep/kernel.common.timing
* tests/kernel/tickless/tickless_concept/kernel.tickless.concept
* tests/kernel/workq/work_queue/kernel.workqueue

The first change is related to the simulation itself and reduces the CLINT frequency in Renode model down to 16 MHz to correspond with the oscillator frequency given by the FE310's datasheet and the HiFive1 board schematic. This fixes the first two tests.

The second change is reducing the MIN_DELAY define to 100. This causes the RISC-V machine timer driver to update the `mtimecmp` register more often, which in turn addresses the `work_queue/kernel.workqueue` problem with work items finishing prematurely, causing the above-mentioned test to fail.

### CODE REVIEW

File: drivers/timer/riscv_machine_timer.c [hunk 1]
```diff
@@ -14,7 +14,7 @@
 				 / (uint64_t)CONFIG_SYS_CLOCK_TICKS_PER_SEC))
 #define MAX_CYC INT_MAX
 #define MAX_TICKS ((MAX_CYC - CYC_PER_TICK) / CYC_PER_TICK)
-#define MIN_DELAY 1000
+#define MIN_DELAY 100
```
Comment (edersondisouza) [inline drivers/timer/riscv_machine_timer.c:17]: I wonder if this shouldn't become a config option, then.

Comment (fkokosinski) [reply drivers/timer/riscv_machine_timer.c:17]: Sure, good idea.

**Comment by fkokosinski** [comment]
I'm not entirely sure why the following tests are failing on the CI:
* tests/kernel/timer/timer_api/kernel.timer.tickless
* tests/kernel/tickless/tickless_concept/kernel.tickless.concept
* tests/kernel/mem_slab/mslab_concept/kernel.memory_slabs.concept
* tests/kernel/timer/timer_api/kernel.timer

I've retested them on HW, individually with `west run` and by using `twister` with `-p hifive1` and they all pass on my local machine. They also work properly in a container that uses the developer docker image from the [docker-image](https://github.com/zephyrproject-rtos/docker-image) repository.

**Review by edersondisouza — DISMISSED** [review]
LGTM

And I can confirm that tests are passing locally on renode. Tried re-run the failed jobs to see if it wasn't a fluke, but it seems the same tests failed again. Weird.

Time to update renode on CI? I think a new version was released last week...

**Comment by fkokosinski** [comment]
I've recreated this behavior locally, it seems that I've been using the docker image wrongly. Bumping Renode version solved the issue for me and all tests are passing with Renode 1.13.0.

I've created https://github.com/zephyrproject-rtos/docker-image/pull/102 to address this problem and bump Renode version.

**Comment by fkokosinski** [comment]
> I've created [zephyrproject-rtos/docker-image#102](https://github.com/zephyrproject-rtos/docker-image/pull/102) to address this problem and bump Renode version.

I'll re-run the pipeline as soon as #45668 (which bumps the container used by the CI) is merged.

**Comment by fkokosinski** [comment]
@edersondisouza Hi, all tests are now passing. Could you reapprove this PR?

**Review by tgorochowik — APPROVED** [review]

**Review by edersondisouza — APPROVED** [review]

**Review by katsuster — APPROVED** [review]

**Pull request merged**